### ISSUE TITLE
TypeError: Cannot read property 'start' of undefined

### ISSUE THREAD

**Issue by niftylettuce** [stylelint/stylelint#654]
TypeError: Cannot read property 'start' of undefined
    at AtRule.positionBy (/Users/nexus/Public/standard-signature/node_modules/postcss/lib/node.js:249:30)
    at exports.default (/Users/nexus/Public/standard-signature/node_modules/stylelint/dist/utils/report.js:46:32)
    at /Users/nexus/Public/standard-signature/node_modules/stylelint/dist/rules/at-rule-empty-line-before/index.js:61:25
    at /Users/nexus/Public/standard-signature/node_modules/postcss/lib/container.js:141:34
    at /Users/nexus/Public/standard-signature/node_modules/postcss/lib/container.js:73:26
    at Root.each (/Users/nexus/Public/standard-signature/node_modules/postcss/lib/container.js:60:22)
    at Root.walk (/Users/nexus/Public/standard-signature/node_modules/postcss/lib/container.js:72:21)
    at Root.walkAtRules (/Users/nexus/Public/standard-signature/node_modules/postcss/lib/container.js:139:25)
    at /Users/nexus/Public/standard-signature/node_modules/stylelint/dist/rules/at-rule-empty-line-before/index.js:25:10
    at /Users/nexus/Public/standard-signature/node_modules/stylelint/dist/postcssPlugin.js:120:80
    at Array.forEach (native)
    at /Users/nexus/Public/standard-signature/node_modules/stylelint/dist/postcssPlugin.js:101:33
    at run (/Users/nexus/Public/standard-signature/node_modules/core-js/modules/es6.promise.js:104:47)
    at /Users/nexus/Public/standard-signature/node_modules/core-js/modules/es6.promise.js:115:28
    at flush (/Users/nexus/Public/standard-signature/node_modules/core-js/modules/$.microtask.js:19:5)
    at nextTickCallbackWith0Args (node.js:433:9)
    at process._tickCallback (node.js:362:13)

**Comment by niftylettuce** [comment]
Here's the order of my middleware:

```js
// postcss middleware
let plugins = [
  cssImport(),
  assets({
    basePath: path.join(__dirname, '..', 'static-assets'),
    loadPaths: [
      path.join(__dirname, '..', 'static-assets')
    ]
  }),
  simpleVars(),
  animation(),
  cssnext(),
  hide(),
  rucksack(),
  nested(),
  at2x(),
  stylelint(),
  colorguard(),
  reporter({
    clearMessages: true
  })
  //cssnano()
];
```

**Comment by davidtheclark** [comment]
@niftylettuce : Please try linting your CSS *before* applying the transformations. That's generally what we recommend --- after all, you want to lint the original source, not the transformed code.

**Comment by jeddy3** [comment]
@davidtheclark I think we said we'd look into tweaking the docs if this popped up again. Perhaps we can expand step 4 in the [Quick start](https://github.com/stylelint/stylelint/blob/master/README.md#quick-start) to something like?:

"Decide whether to use the CLI, Node API, or PostCSS plugin. If you use the PostCSS plugin we recommend you lint your CSS _before_ applying any transformations. You can do this by either placing stylelint at the beginning of your plugin pipeline or by creating a separate linting process that is independent of your build one."

Too wordy?

**Comment by davidtheclark** [comment]
Yes, sounds great. And hey your writing is swell!

**Comment by jeddy3** [comment]
Closing this issue.

@niftylettuce Feel free to reopen it if the advice [here](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/postcss-plugin.md#usage-examples) doesn't work for you.

**Comment by tiagocpontesp** [comment]
I'm using [rollup-plugin-postcss](https://github.com/egoist/rollup-plugin-postcss). If alongside [postcss-modules](https://github.com/css-modules/postcss-modules), regardless of the order, I get this error. Maybe this should be reopened.

**Comment by tiagocpontesp** [comment]
Just for reference, disabling the rule does the job. (this override isn't included in the standard css-modules config).
`
    'property-no-unknown': {
      ignoreProperties: [
        'start',
      ],
    },`
